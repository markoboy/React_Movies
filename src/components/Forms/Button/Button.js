import clsx from 'clsx';
import PropTypes from 'prop-types';
import React, { useMemo } from 'react';
import './Button.scss';

/**
 * @param {Object} props React props
 * @param {'button' | 'submit' | 'reset'} [props.type] The type of the button
 * @param {string} [props.classes] Any classes to be appended in the button as modifiers
 */
export default function Button({
  children,
  type,
  classes,
  onClick,
}) {
  const className = useMemo(() => clsx('btn', classes), [classes]);

  return (
    <button type={type} className={className} onClick={onClick}>
      {children}
    </button>
  );
}

Button.propTypes = {
  type: PropTypes.oneOf(['button', 'submit', 'reset']),
  classes: PropTypes.string,
  onClick: PropTypes.func,
};

Button.defaultProps = {
  type: 'button',
  classes: '',
  onClick: null,
};
