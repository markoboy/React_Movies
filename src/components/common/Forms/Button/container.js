import clsx from 'clsx';
import React, { useMemo } from 'react';
import ButtonComponent from './component';

function ButtonContainer({ children, type, classes, tabIndex, onClick }) {
  const className = useMemo(() => clsx('btn', classes), [classes]);

  return (
    <ButtonComponent
      type={type}
      classes={className}
      onClick={onClick}
      tabIndex={tabIndex}
    >
      {children}
    </ButtonComponent>
  );
}

ButtonContainer.defaultProps = ButtonComponent.defaultProps;

ButtonContainer.propTypes = ButtonComponent.propTypes;

export default ButtonContainer;
