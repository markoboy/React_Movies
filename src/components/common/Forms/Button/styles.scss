@import 'essential';

.btn {
  outline: none;
  border: none;
  padding: $form-input-padding;
  border-radius: $border-radius;
  font-size: $form-input-font-size;
  text-transform: uppercase;
  cursor: pointer;
  transition: background-color 0.3s linear, color 0.3s linear;

  &:focus,
  &:hover {
    box-shadow: $box-shadow;
  }

  &--primary {
    background-color: $color-primary;
    color: $color-white;

    &:focus,
    &:hover {
      background-color: $color-white;
      color: $color-primary;
    }
  }

  &--secondary {
    background-color: rgba($color-secondary, 0.4);
    color: $color-primary;

    &:focus,
    &:hover {
      background-color: $color-primary;
      color: $color-secondary;
    }
  }

  &--outline {
    background-color: transparent;
    color: $color-primary;
    border: 1px solid $color-primary;

    &:focus,
    &:hover {
      background-color: $color-primary;
      color: $color-white;
    }
  }

  &--with-icon {
    > img,
    > svg {
      margin-right: 0.5em;
    }
  }

  &--full-width {
    width: 100%;
  }
}
