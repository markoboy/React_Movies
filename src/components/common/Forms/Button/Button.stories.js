import React from 'react';
import { faPlus } from '@fortawesome/free-solid-svg-icons/faPlus';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import withStoryBackgroundWrapper from '@components/hocs/WithStoryBackgroundWrapper';
import Button from './index';

export default {
  component: Button,
  title: 'Common/Forms/Button',
  decorators: [withStoryBackgroundWrapper],
};

const Template = (args) => <Button {...args} />;

export const Default = Template.bind({});
Default.args = {
  children: 'Button',
};

export const Primary = Template.bind({});
Primary.args = {
  classes: 'btn--primary',
  children: 'Button',
};

export const Secondary = Template.bind({});
Secondary.args = {
  classes: 'btn--secondary',
  children: 'Button',
};

export const Outline = Template.bind({});
Outline.args = {
  classes: 'btn--outline',
  children: 'Button',
};

export const FullWidth = Template.bind({});
FullWidth.args = {
  classes: 'btn--full-width',
  children: 'Button',
};

export const WithIcon = Template.bind({});
WithIcon.args = {
  classes: 'btn--with-icon',
  children: (
    <>
      <FontAwesomeIcon size="xs" icon={faPlus} />
      Button
    </>
  ),
};
