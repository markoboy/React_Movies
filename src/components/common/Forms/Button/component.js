import clsx from 'clsx';
import PropTypes from 'prop-types';
import React, { useMemo } from 'react';
import './styles.scss';

/**
 * @param {Object} props React props
 * @param {'button' | 'submit' | 'reset'} [props.type] The type of the button
 * @param {string} [props.classes] Any classes to be appended in the button as modifiers
 */
export default function ButtonComponent({
  children,
  type,
  classes,
  onClick,
}) {
  const className = useMemo(() => clsx('btn', classes), [classes]);

  return (
    <button type={type} className={className} onClick={onClick}>
      {children}
    </button>
  );
}

ButtonComponent.propTypes = {
  type: PropTypes.oneOf(['button', 'submit', 'reset']),
  classes: PropTypes.string,
  onClick: PropTypes.func,
};

ButtonComponent.defaultProps = {
  type: 'button',
  classes: '',
  onClick: null,
};
