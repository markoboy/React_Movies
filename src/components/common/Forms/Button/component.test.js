import React from 'react';
import { render } from '@testing-library/react';

import Button from './component';

describe('Button Component', () => {
  it('renders Button Component with provided text', () => {
    const { container } = render(<Button>Button component</Button>);

    expect(container).toMatchSnapshot();
  });
});
