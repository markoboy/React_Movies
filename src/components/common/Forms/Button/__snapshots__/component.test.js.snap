// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component renders Button Component with provided text 1`] = `
<div>
  <button
    class=""
    tabindex="0"
    type="button"
  >
    Button component
  </button>
</div>
`;
